### PR TITLE
Add worker-src to CSP

### LINKED ISSUE #11035: CSP child-src is deprecated
We should update our CSP accordingly and use only use `worker-src` and `frame-src`

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #11035

### CODE REVIEW

**Comment by skjnldsv** [comment]
I have no idea how to review this :/
Any hint? :)

**Review by ChristophWurst — DISMISSED** [review]

File: lib/public/AppFramework/Http/EmptyContentSecurityPolicy.php [hunk 1]
```diff
@@ -355,6 +359,30 @@ public function disallowFrameAncestorDomain($domain) {
...
+
+	/**
+	 * Remove domain from which workers can be loaded
+	 *
+	 * @param string $domain
+	 * @return $this
+	 * @since 15.0.0
+	 */
+	public function disallowWorkerSrcDomain(string $domain) {
+		$this->allowedWorkerSrcDomains[] = array_diff($this->allowedWorkerSrcDomains, [$domain]);
```
[the remark this answers is not in the document]
Comment (rullzer) [reply lib/public/AppFramework/Http/EmptyContentSecurityPolicy.php]: woeps

**Comment by rullzer** [comment]
@skjnldsv it is mostly following the code. The CSP is pretty straight forward.

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
Changes looks good and kinda make sense :elephant:

**Pull request merged**